packages/app-desktop/gui/NoteEditor/NoteEditorWrapper.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import { useCallback, useMemo, useState } from 'react';
+import NoteEditor from './NoteEditor';
+import StyleSheetContainer from '../StyleSheets/StyleSheetContainer';
+import { connect } from 'react-redux';
+import { AppState } from '../../app.reducer';
+import { Dispatch } from 'redux';
+import NewWindowOrIFrame from '../NewWindowOrIFrame';
+import WindowCommandHandler from '../WindowCommandHandler/WindowCommandHandler';
+
+import createCache from '@emotion/cache';
+import { CacheProvider } from '@emotion/react';
+const { StyleSheetManager } = require('styled-components');
+
+interface Props {
+	dispatch: Dispatch;
+	themeId: number;
+	bodyEditor: string;
+	newWindow: boolean;
+	noteId?: string;
+	defaultNoteId: string;
+	provisionalNoteIds: string[];
+}
+
+const NoteEditorWrapper: React.FC<Props> = props => {
+	const onClose = useCallback(() => {
+		props.dispatch({ type: 'NOTE_WINDOW_CLOSE', noteId: props.noteId });
+	}, [props.dispatch, props.noteId]);
+
+	const noteId = props.noteId ?? props.defaultNoteId;
+	const editor = <NoteEditor bodyEditor={props.bodyEditor} noteId={noteId} isProvisional={props.provisionalNoteIds.includes(noteId)}/>;
+
+	const windowContent = <NewWindowOrIFrame newWindow={props.newWindow} onClose={onClose}>
+		<LibraryStyleRoot>
+			<WindowCommandHandler />
+			{editor}
+		</LibraryStyleRoot>
+		<StyleSheetContainer />
+	</NewWindowOrIFrame>;
+	return props.newWindow ? windowContent : editor;
+};
+
+interface StyleProviderProps {
+	children: React.ReactNode[]|React.ReactNode;
+}
+
+// Sets the root style container for libraries. At present, this is needed by react-select (which uses @emotion/...)
+// and styled-components.
+// See: https://github.com/JedWatson/react-select/issues/3680 and https://github.com/styled-components/styled-components/issues/659
+const LibraryStyleRoot: React.FC<StyleProviderProps> = props => {
+	const [dependencyStyleContainer, setDependencyStyleContainer] = useState<HTMLDivElement|null>(null);
+	const cache = useMemo(() => {
+		return createCache({
+			key: 'new-window-cache',
+			container: dependencyStyleContainer,
+		});
+	}, [dependencyStyleContainer]);
+
+	return <>
+		<div ref={setDependencyStyleContainer}></div>
+		<StyleSheetManager target={dependencyStyleContainer}>
+			<CacheProvider value={cache}>
```

Comment:
This code uses `CacheProvider` from `@emotion/react` to cause `react-select` to insert its styles into the new window (see [comment](https://github.com/JedWatson/react-select/issues/3680#issuecomment-809518245)). `@emotion/react` is a dependency of `react-select` and, at present, is not a direct dependency of Joplin.

Adding `@emotion/react` to `app-desktop/package.json` could be problematic if `react-select` is updated without also updating the `@emotion/react` version &mdash; having multiple versions of `@emotion/react` [may cause styling issues](https://github.com/emotion-js/emotion/issues/2383) (or perhaps break styling just in new windows).

See [this unmerged upstream pull request](https://github.com/JedWatson/react-select/pull/5939) for a possible future solution that doesn't involve `import`ing transitive dependencies.